src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -2692,4 +2691,110 @@ public void terminate() throws Exception {
         }
     }
+
+    @Nested
+    @Tag(Constants.xAzureSQLDW)
+    public class BigDecimalPrecisionTest {
+
+    	private final String procName1 = AbstractSQLGenerator
+    			.escapeIdentifier(RandomUtil.getIdentifier("test_bigdecimal_3"));
+    	private final String procName2 = AbstractSQLGenerator
+    			.escapeIdentifier(RandomUtil.getIdentifier("test_bigdecimal_5"));
+    	private final String procNameMaxScale = AbstractSQLGenerator
+    			.escapeIdentifier(RandomUtil.getIdentifier("test_bigdecimal_max_scale"));
+    	private final String procNameMaxPrecision = AbstractSQLGenerator
+    			.escapeIdentifier(RandomUtil.getIdentifier("test_bigdecimal_max_precision"));
+
+    	@BeforeEach
+    	public void init() throws SQLException {
+    		try (Connection connection = getConnection()) {
+    			String dropProcedureSQL = "DROP PROCEDURE IF EXISTS " + procName1 + ", " + procName2 + ", "
+    					+ procNameMaxScale + ", " + procNameMaxPrecision;
+    			try (Statement stmt = connection.createStatement()) {
+    				stmt.execute(dropProcedureSQL);
+    			}
+
+    			String createProcedureSQL1 = "CREATE PROCEDURE " + procName1 + "\n"
+    					+ "    @big_decimal_type      decimal(15, 3),\n"
+    					+ "    @big_decimal_type_o    decimal(15, 3) OUTPUT\n" + "AS\n" + "BEGIN\n"
+    					+ "    SET @big_decimal_type_o = @big_decimal_type;\n" + "END;";
+    			String createProcedureSQL2 = "CREATE PROCEDURE " + procName2 + "\n"
+    					+ "    @big_decimal_type      decimal(15, 5),\n"
+    					+ "    @big_decimal_type_o    decimal(15, 5) OUTPUT\n" + "AS\n" + "BEGIN\n"
+    					+ "    SET @big_decimal_type_o = @big_decimal_type;\n" + "END;";
+    			String createProcedureMaxScale = "CREATE PROCEDURE " + procNameMaxScale + "\n"
+    					+ "    @big_decimal_type      decimal(38, 38),\n"
+    					+ "    @big_decimal_type_o    decimal(38, 38) OUTPUT\n" + "AS\n" + "BEGIN\n"
+    					+ "    SET @big_decimal_type_o = @big_decimal_type;\n" + "END;";
+    			String createProcedureMaxPrecision = "CREATE PROCEDURE " + procNameMaxPrecision + "\n"
+    					+ "    @big_decimal_type      decimal(38, 0),\n"
+    					+ "    @big_decimal_type_o    decimal(38, 0) OUTPUT\n" + "AS\n" + "BEGIN\n"
+    					+ "    SET @big_decimal_type_o = @big_decimal_type;\n" + "END;";
+
+    			try (Statement stmt = connection.createStatement()) {
+    				stmt.execute(createProcedureSQL1);
+    				stmt.execute(createProcedureSQL2);
+    				stmt.execute(createProcedureMaxScale);
+    				stmt.execute(createProcedureMaxPrecision);
+    			}
+    		}
+    	}
+
+    	@AfterEach
+    	public void terminate() throws SQLException {
+    		try (Connection connection = getConnection()) {
+    			try (Statement stmt = connection.createStatement()) {
+    				String dropProcedureSQL = "DROP PROCEDURE IF EXISTS " + procName1 + ", " + procName2 + ", "
+    						+ procNameMaxScale + ", " + procNameMaxPrecision;
+    				stmt.execute(dropProcedureSQL);
+    			}
+    		}
```

Comment:
There's a utility function to do this: `TestUtils.dropProcedureIfExists`.

Although, now that we no longer need to support SQL Server 2014, the `TestUtils.dropXXXIfExists` methods could be enhanced to use the `IF EXISTS` syntax and even take an array of strings to drop multiple objects in one command. But this enhancement is optional.